src/main/java/frc/utils/digitalinput/channeled/ChanneledDigitalInput.java

```diff
@@ -0,0 +1,31 @@
+package frc.utils.digitalinput.channeled;
+
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.utils.digitalinput.DigitalInputInputsAutoLogged;
+import frc.utils.digitalinput.IDigitalInput;
+
+public class ChanneledDigitalInput implements IDigitalInput {
+
+	private final DigitalInput digitalInput;
+	private final Debouncer debouncer;
+	private final boolean normallyClosed;
+
+	public ChanneledDigitalInput(int channel, double debounceTime, boolean normallyClosed) {
```

Comment:
Rename boolean is normally closed to invert (sometimes you just jeednto invert the value, one of them is nirmally closed/open)
+ what about the MXP ports?

Reply:
they are included in the digital input and have different channel  numbers